src/auth/User.cc

```diff
@@ -125,104 +79,43 @@ Auth::User::~User()
     /* free cached acl results */
     aclCacheMatchFlush(&proxy_match_cache);
 
-    /* free seen ip address's */
-    clearIp();
-
     if (username_)
         xfree((char*)username_);
 
     /* prevent accidental reuse */
     auth_type = Auth::AUTH_UNKNOWN;
 }
 
+/// generate the cache key for an ipList entry
+static const SBuf
+BuildIpKey(const Ip::Address &ip)
+{
+    SBuf key;
+    auto *buf = key.rawAppendStart(MAX_IPSTRLEN);
+    const auto len = ip.toHostStr(buf, MAX_IPSTRLEN);
+    key.rawAppendFinish(buf, len);
+    return key;
```

Comment:
TBH I feel like this would belong better to the Ip::Address class.

Reply:
FWIW, I recommend addressing my change request about ipList key type before resuming work on this function.